node_modules/oae-principals/lib/rest.user.js

```diff
@@ -566,6 +567,63 @@ OAE.tenantRouter.on('post', '/api/user/:userId/picture', function(req, res) {
 });
 
 /**
+ * @REST postUserEmailVerify
+ *
+ * Verify an email token
+ *
+ * @Server      tenant
+ * @Method      POST
+ * @Path        /user/email/verify
+ * @FormParam   {string}        email                       The email to verify
+ * @FormParam   {string}        token                       The token to verify
+ * @HttpResponse                200                         The token was valid and the email address for the user that was associated with it is verified
+ * @HttpResponse                400                         A token was not specified
+ * @HttpResponse                404                         The token was not associated with a user
+ */
+OAE.tenantRouter.on('post', '/api/user/email/verify', function(req, res) {
+    PrincipalsAPI.verifyEmail(req.ctx, req.body.email, req.body.token, function(err, user) {
+        if (err) {
+            return res.send(err.code, err.msg);
+        }
+
+        // If the user is anonymous we log them into the system
+        if (!req.ctx.user()) {
+            return AuthenticationUtil.handleLogin('local', user, req, res);
```

Comment:
I don't know if I agreed that we should do this before, but at the moment I'm considering the case where someone has entered an incorrect email when updating their profile, and in that context this seems really risky.